hamilton/execution/executors.py

```diff
@@ -97,12 +97,21 @@ def base_execute_task(task: TaskImplementation) -> Dict[str, Any]:
         if not getattr(node_, "callable_modified", False):
             node_._callable = _modify_callable(node_.node_role, node_.callable)
         setattr(node_, "callable_modified", True)
-    return execute_subdag(
+    out = execute_subdag(
         nodes=task.nodes,
         inputs=task.dynamic_inputs,
         adapter=task.adapters[0],  # TODO -- wire through multiple graph adapters
         overrides={**task.dynamic_inputs, **task.overrides},
     )
+    # This selection is for GC
+    # We also need to get the override values
+    # This way if its overridden we can ensure it gets passed to the right one
+    final_retval = {
+        key: value
+        for key, value in out.items()
+        if key in task.outputs_to_compute or key in task.overrides
+    }
+    return final_retval
```

Comment:
Yeah you just need to update the doc string in this function to say you're only returning the subset of nodes that are in the output or are part of overrides.